File: my_code.js
const { InfluxDB } = require("@influxdata/influxdb-client");

// You can generate a Token from the "Tokens Tab" in the UI
const token =
  "7wQejxjJ5FVj4IpXU2SdgAOngBvjAc8Nro3QcbsDj6Q1A7rwB5zWP6wxqge2v8DVlb5PXIFOBTOUg6viiStuTw==";
const org = "my-org";
const bucket = "my-bucket-1";

const client = new InfluxDB({ url: "http://localhost:8086", token: token });
const { Point } = require("@influxdata/influxdb-client");
const writeApi = client.getWriteApi(org, bucket);
writeApi.useDefaultTags({ host: "host2" });

const point = new Point("mem").floatField("used_percent", Math.random());
// writeApi.writePoint(point);
writeApi
  .close()
  .then(() => {
    console.log("FINISHED");
  })
  .catch((e) => {
    console.error(e);
    console.log("\\nFinished ERROR");
  });

const queryApi = client.getQueryApi(org);

const query = `from(bucket: \"${bucket}\") |> range(start: -1h)`;
queryApi.queryRows(query, {
  next(row, tableMeta) {
    const o = tableMeta.toObject(row);
    console.log(
      `${o._time} ${o._measurement} host: ${o.host} in \'${o.location}\' (${o.example}): ${o._field}=${o._value}`
    );
  },
  error(error) {
    console.error(error);
    console.log("nFinished ERROR");
  },
  complete() {
    console.log("Finished SUCCESS");
  },
});

File: my_code.rb
# require 'influxdb-client'

# # You can generate a Token from the "Tokens Tab" in the UI
# token = '7wQejxjJ5FVj4IpXU2SdgAOngBvjAc8Nro3QcbsDj6Q1A7rwB5zWP6wxqge2v8DVlb5PXIFOBTOUg6viiStuTw=='
# org = 'my-org'
# bucket = 'my-bucket-1'

# client = InfluxDB2::Client.new('http://localhost:8086', token,
#                                precision: InfluxDB2::WritePrecision::NANOSECOND)

# # write_api = client.create_write_api
# # # data
# # data = 'mem,host=host1 used_percent=23.43234543'
# # write_api.write(data: data, bucket: bucket, org: org)
# # #####################################################
# # # point
# # point = InfluxDB2::Point.new(name: 'mem')
# #                         .add_tag('host', 'host1')
# #                         .add_field('used_percent', 23.43234543)
# #                         .time(Time.now.utc, InfluxDB2::WritePrecision::NANOSECOND)
# # write_api.write(data: point, bucket: bucket, org: org)
# # #####################################################
# # # hash
# # hash = { name: 'h2o',
# #          tags: { host: 'aws', region: 'us' },
# #          fields: { level: 5, saturation: '99%' },
# #          time: Time.now.utc }
# # write_api.write(data: hash, bucket: bucket, org: org)
# # #####################################################
# # # Batch Sequence to write data
# # point = InfluxDB2::Point.new(name: 'mem')
# #                         .add_tag('host', 'host1')
# #                         .add_field('used_percent', 23.43234543)
# #                         .time(Time.now.utc, InfluxDB2::WritePrecision::NANOSECOND)

# # hash = { name: 'h2o',
# #          tags: { host: 'aws', region: 'us' },
# #          fields: { level: 5, saturation: '99%' },
# #          time: Time.now.utc }
# # data = 'mem,host=host1 used_percent=23.23234543'
# # write_api.write(data: [point, hash, data], bucket: bucket, org: org)
# # #####################################################
# # query
# query = 'from(bucket: "' + bucket + '") |> range(start: -1h)'

# puts query
# tables = client.create_query_api.query(query: query, org: org)

# puts tables
